### PR TITLE
Notebooks

### PR DESCRIPTION
**Pull request by salsakran**
Adding some notebooks for common backend repl commands. 

The intention is that this will be a repository of things I continuously forget, as well as a form of runnable documentation for common backend tasks.

### CODE REVIEW

**Comment by salsakran** [comment]
@camsaul could you help out @sbelak and myself by filling in the bits of `common_data_queries` we're stuck on?

**Comment by camsaul** [comment]
You're stuck on *all* of those questions? I'd recommend you guys do a quick read-through of [Toucan DB functions guide](https://github.com/metabase/toucan/blob/master/docs/db-functions.md) and that should explain 95% of the questions you have listed. Besides that all you need to do when you launch the REPL is

```clojure
(require 'metabase.db)
(metabase.db/setup-db!)
;; ...
(toucan.db/select 'Table) ; get all the tables (etc.)
```

**Comment by salsakran** [comment]
@camsaul That was a quick dump of questions. Simon will have others crop up. 

Do me a favor and help out here, as while a lot of this stuff is obvious for you, it's not exactly obvious for the rest of us. RTFM is not a helpful response.

**Comment by camsaul** [comment]
# how do I fetch a table by id?

```clojure
(toucan.db/select-one 'Table :id 1)
```

# how do I get all a tables rows?

```clojure
(metabase.query-processor/process-query
  {:database 1
   :type :query
   :query {:source-table 1}})
```

# how do I get a sample of a tables rows?

```clojure
(metabase.query-processor/process-query
  {:database 1
   :type :query
   :query {:source-table 1
           :limit 10}})
```


# how do I get all the fields of a table?

```clojure
(toucan.db/select 'Field :table_id 1)
```

# how do I fetch a field by id?

```clojure
(toucan.db/select-one 'Field :id 1)
```

# how do I fetch a sample of values for a field

```clojure
(metabase.query-processor/process-query
  {:database 1
   :type :query
   :query {:fields [[:field-id 1]]
           :source-table 1
           :limit 10}})
```

# how do I fetch all the values + their frequencies (eg, count grouped by field)

```clojure
(metabase.query-processor/process-query
  {:database 1
   :type :query
   :query {:aggregation [:count]
           :breakout [[:field-id 1]]
           :source-table 1
           :limit 10}})
```


# how do I fetch a metric by id?

```clojure
(toucan.db/select-one 'Metric :id 1)
```

# how do I calculate the metric

huh?

# how do I get all dimensions (eg all attached fields) for a metric?

```clojure
(toucan.db/select-one-field :definition 'Metric :id 1)
```

# how do I get the type of each dimension?

By "dimension" do you mean a Field?

```clojure
(toucan.db/select-one-field :base_type 'Field :id 1)
```

# how do I get a metric broken out by one of its dimensions?

Not 100% sure I understand what you're asking :/

# how do I filter a matric on one of its dimensions?

Not 100% sure I understand what you're asking :/

**Comment by camsaul** [comment]
I'd highly recommend taking a look at the [MBQL reference](https://github.com/metabase/metabase/wiki/Query-Language-'98) for a more detailed explanation of how everything fits together as well as the Toucan DB functions guide I linked to above

**Comment by salsakran** [comment]
@metabase/core-developers do we want to merge this? I'm in favor, as I've found notebooks pretty useful in sharing info w/ @sbelak

File: project.clj [hunk 1]
```diff
@@ -81,9 +81,17 @@
...
                  [toucan "1.0.3"                                      ; Model layer, hydration, and DB utilities
-                  :exclusions [honeysql]]]
+                  :exclusions [honeysql]]
+
+                 ; Notebook utilities and prototypes
+                 [bigml/histogram "4.1.3"]
+                 [bigml/sketchy "0.4.1"]
+                 [net.cgrand/xforms "0.9.2"]
+                 [kixi/stats "0.3.8"]
+                 [redux "0.1.4"]]
```
Comment (camsaul) [inline project.clj]: if these are only used for dev notebooks (i.e. the don't need to be included with the built uberjar) we should make them `:dev` dependencies instead

File: notebooks/common_data_queries [hunk 2]
```diff
@@ -0,0 +1,167 @@
+;; gorilla-repl.fileformat = 1
+
+;; **
+;;; # Metabase Common MBQL forms
+;;; 
+;;; This is a document to simplify working with MBQL for data science-y types as well as people like me (Sameer) with no memory
+;; **
+
+;; @@
+(ns parched-sanctuary
```
Comment (camsaul) [inline notebooks/common_data_queries]: parched-sanctuary ?

**Comment by camsaul** [comment]
To clarify, the purpose of this is just to help remember some common dev REPL commands? Maybe this would be better (easier to edit) in a wiki page instead of in the codebase

**Comment by camsaul** [comment]
We can stick this in a wiki page

**Pull request closed without merging**